discord-bot/src/index.ts

```diff
@@ -0,0 +1,202 @@
+
+import { REST } from '@discordjs/rest'
+import * as console from 'console'
+import {
+  ChatInputCommandInteraction,
+  Client,
+  Collection,
+  EmbedBuilder,
+  Events,
+  GatewayIntentBits,
+  MessageReaction,
+  PartialMessageReaction,
+  Partials,
+  PartialUser,
+  Routes,
+  SlashCommandBuilder,
+  TextChannel,
+  User,
+} from 'discord.js'
+import * as fs from 'fs'
+import * as path from 'path'
+import * as process from 'process'
+import { fileURLToPath } from 'url'
+import { config } from './constants/config.js'
+import { getAnyHandledEmojiKey } from './emojis.js'
+import {
+  getMarketFromSlug,
+  getSlug,
+  handleReaction,
+  sendThreadErrorMessage,
+} from './helpers.js'
+import { registerApiKey } from './register-api-key.js'
+import { startListener } from './server.js'
+import { messagesHandledViaInteraction, userApiKey } from './storage.js'
+
+const { id: clientId } = config.client
+const token = process.env.DISCORD_BOT_TOKEN
+if (!token) throw new Error('No DISCORD_BOT_TOKEN env var set.')
+export const commands = new Collection<
+  string,
+  {
+    data: SlashCommandBuilder
+    execute: (interaction: ChatInputCommandInteraction) => Promise<any>
+  }
+>()
+const readCommandsFromFiles = async () => {
+  // Load commands
+  console.log('Loading commands... ')
+  const commandsPath = path.join(
+    path.dirname(fileURLToPath(import.meta.url)),
+    'commands'
+  )
+  const extension = commandsPath.includes('lib') ? '.js' : '.ts'
+  console.log('Commands path: ', commandsPath)
+  const commandFiles = fs
+    .readdirSync(commandsPath)
+    .filter((file: string) => file.endsWith(extension))
+
+  for (const file of commandFiles) {
+    const filePath = path.join(commandsPath, file)
+    const command = await import(filePath)
+    // Set a new item in the Collection
+    // With the key as the command name and the value as the exported module
+    commands.set(command.data.name, command)
+  }
+  console.log('Read commands from files')
+}
+await readCommandsFromFiles()
+
+const rest = new REST({ version: '10' }).setToken(token)
+
+console.log('Logging in... with client id: ', clientId)
+export const client = new Client({
+  intents: [
+    GatewayIntentBits.Guilds,
+    GatewayIntentBits.GuildMessages,
+    GatewayIntentBits.GuildMessageReactions,
+    GatewayIntentBits.DirectMessages,
+  ],
+  partials: [
+    Partials.Message,
+    Partials.Channel,
+    Partials.Reaction,
+    Partials.User,
+  ],
+})
+client.once(Events.ClientReady, () => {
+  console.log('Ready!')
+})
+
+client.on(Events.InteractionCreate, async (interaction) => {
+  if (!interaction.isChatInputCommand()) return
+
+  const command = commands.get(interaction.commandName)
+  if (!command) return
+
+  try {
+    await command.execute(interaction)
+  } catch (error) {
+    console.error(error)
+    await interaction.reply({
+      content: 'There was an error while executing this command :(',
+      ephemeral: true,
+    })
+  }
+})
+
+client.login(token)
```

Comment:
I think you should `await` this before trying to do stuff like refresh the slash commands.